sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -117,7 +118,9 @@ case class DataSourceV2Relation(
 case class DataSourceV2ScanRelation(
     relation: DataSourceV2Relation,
     scan: Scan,
-    output: Seq[AttributeReference]) extends LeafNode with NamedRelation {
+    output: Seq[AttributeReference],
+    distribution: Distribution = UnspecifiedDistribution,
```

Comment:
hmm, ideally a scan node should report output partitioning, the parent node such as join node reports required distribution. This looks a bit counterintuitive.

Reply:
Yea I agree with you. What do you think if we pass the clustering expressions instead? something like:

```scala
distribution: Option[Seq[Expression]] = None,
```

Reply:
Can we just use `Partitioning` here? Which can be `DataSourceHashPartitioning` under the hood.

Reply:
I'm using 
```scala
clustering: Option[Seq[Expression]] = None
```

for now, since `DataSourceHashPartitioning` is the `outputPartitioning` we construct inside `DataSourceV2ScanExecBase`, so it isn't available at this point.

Let me know if you have any other thought.